### PR TITLE
feat: add support for avail as a da

### PR DESCRIPTION
**Pull request by mtsitrin**
# PR Standards

## Opening a pull request should be able to meet the following requirements

---

For Author:

- [ ]  Targeted PR against correct branch
- [ ]  included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Linked to Github issue with discussion and accepted design
- [ ]  Targets only one github issue
- [ ]  Wrote unit and integration tests
- [ ]  All CI checks have passed
- [ ]  Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code)

---

For Reviewer:

- [ ]  confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Reviewers assigned
- [ ]  confirmed all author checklist items have been addressed

---

After reviewer approval:

- [ ]  In case targets main branch, PR should be squashed and merged.
- [ ]  In case PR targets a release branch, PR should be rebased.

### CODE REVIEW

File: cmd/keys/export/export.go [hunk 1]
```diff
@@ -40,10 +41,13 @@ func Cmd() *cobra.Command {
 				out, err := utils.ExecBashCommand(exportKeyCmd)
 				printKeyOutput(out.String(), err)
 			} else if keyID != "" && keyID == damanager.GetKeyName() {
+				//TODO: avail doesn't need cmd to get the keys, it's stored in the config
```
Comment (ItayLevyOfficial) [inline cmd/keys/export/export.go:44]: avail doesn't get included in the supported keys in this flow right

File: data_layer/avail/avail.go [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+	if err != nil {
+		return config, err
+	}
+	err = toml.Unmarshal(tomlBytes, &config)
+	if err != nil {
+		return config, err
+	}
+
+	return config, nil
+}
```
Comment (ItayLevyOfficial) [inline data_layer/avail/avail.go:215]: this file is over 200 lines of code, let's try to keep file size around 150 loc

File: data_layer/da_layer.go [hunk 3]
```diff
@@ -36,10 +38,11 @@ func NewDAManager(datype config.DAType, home string) *DAManager {
 		dalayer = &celestia.Celestia{
 			Root: home,
 		}
+	case config.Avail:
+		dalayer = avail.NewAvail(home)
```
Comment (ItayLevyOfficial) [inline data_layer/da_layer.go:42]: maybe break `newavail` to constructor and init config, like celestia?

File: data_layer/damock/damock.go [hunk 4]
```diff
@@ -13,7 +12,7 @@ type DAMock struct {
 }
 
 func (d *DAMock) GetStatus(c config.RollappConfig) string {
-	return ""
+	return "Running mock DA"
```
Comment (ItayLevyOfficial) [inline data_layer/damock/damock.go:15]: why

**Review by ItayLevyOfficial — APPROVED** [review]
LGTM

File: cmd/keys/export/export.go (hunk 1 above)
Comment (mtsitrin) [reply cmd/keys/export/export.go:44]: I know, there should be issue to handle it

File: data_layer/da_layer.go (hunk 3 above)
Comment (mtsitrin) [reply data_layer/da_layer.go:42]: the init of celestia is for the light client. we don't have light client in avail

File: data_layer/damock/damock.go (hunk 4 above)
Comment (mtsitrin) [reply data_layer/damock/damock.go:15]: the DA row is displayed now, u prefer something else to be written for mock DA?

**Review by ItayLevyOfficial — APPROVED** [review]
LGTM

**Pull request merged**